framework/src/media/PlayerWorker.cpp

```diff
@@ -129,59 +129,34 @@ void PlayerWorker::stopWorker()
 	}
 }
 
-void PlayerWorker::startPlayer(std::shared_ptr<MediaPlayerImpl> mp)
+MediaQueue &PlayerWorker::getQueue()
 {
-	medvdbg("MediaPlayer Worker : startPlayer\n");
-	if (mp->mCurState != PLAYER_STATE_READY && mp->mCurState != PLAYER_STATE_PAUSED) {
-		mp->notifyObserver(PLAYER_OBSERVER_COMMAND_ERROR);
-		return;
-	}
-
-	if (mCurPlayer != nullptr) {
-		pausePlayer(mCurPlayer);
-	}
-	mCurPlayer = mp;
-
-	mp->mCurState = PLAYER_STATE_PLAYING;
-	mp->notifyObserver(PLAYER_OBSERVER_COMMAND_STARTED);
+	return mWorkerQueue;
 }
 
-void PlayerWorker::stopPlayer(std::shared_ptr<MediaPlayerImpl> mp)
+void PlayerWorker::increaseRef()
 {
-	medvdbg("MediaPlayer Worker : stopPlayer\n");
-	if (mp->mCurState != PLAYER_STATE_PLAYING && mp->mCurState != PLAYER_STATE_PAUSED) {
-		mp->notifyObserver(PLAYER_OBSERVER_COMMAND_ERROR);
-		return;
-	}
-
-	mCurPlayer = nullptr;
-	mp->mCurState = PLAYER_STATE_READY;
+	mRefCnt++;
 }
 
-void PlayerWorker::pausePlayer(std::shared_ptr<MediaPlayerImpl> mp)
+void PlayerWorker::decreaseRef()
 {
-	medvdbg("MediaPlayer Worker : pausePlayer\n");
-	if (mp->mCurState != PLAYER_STATE_PLAYING) {
-		mp->notifyObserver(PLAYER_OBSERVER_COMMAND_ERROR);
-		return;
-	}
-
-	mCurPlayer = nullptr;
-	mp->mCurState = PLAYER_STATE_PAUSED;
+	mRefCnt--;
 }
 
-MediaQueue &PlayerWorker::getQueue()
+bool PlayerWorker::isAlive()
```

Comment:
In future, you may implement mediaWorker, the parent class of these workers, right?
some places using this function but some places in this place checking mRefCnt >= 0 or mRefCnt == 1, so getWorkerRefCount is more good to expand it later.
What I meaning is that, you had better hide mRefCnt in parent class later, and provice protected method, then every where can use it.
```
class MediaWorker()
{
    private int mRefCnt;
    protected int getWorkerRefCount();
    protected int getWorkerRefCount() {
        return mRefCnt;
    }
}

class Playerworker : MediaWorker
{
    public getPlayerWorkerRefCount();
    public getPlayerWorkerRefCount()
   {
        return super.getWorkerRefCount();
   }
}
```
Of course you can suggest more good logic all the time.

Reply:
Or you can provide some internal api to handle reference count, such as isWorkerRunnable, isWorkerExist